cmd/migrate/list_dynamic_blocks_test.go

```diff
@@ -0,0 +1,279 @@
+package main
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDynamicListTransformation(t *testing.T) {
+	tests := []TestCase{
+		{
+			Name: "dynamic block with for_each should wrap with toset and use each.value",
+			Config: `resource "cloudflare_list" "salesforce_ips" {
+  account_id = "f037e56e89293a057740de681ac9abbe"
+  name       = "salesforce_ips"
+  kind       = "ip"
+
+  dynamic "item" {
+    for_each = local.salesforce_ips
+    content {
+      value {
+        ip = item.value
+      }
+    }
+  }
+}`,
+			Expected: []string{
+				`resource "cloudflare_list" "salesforce_ips" {
+  account_id = "f037e56e89293a057740de681ac9abbe"
+  name       = "salesforce_ips"
+  kind       = "ip"
+}`,
+				`resource "cloudflare_list_item" "salesforce_ips_items" {
+  for_each   = toset(local.salesforce_ips)
+  account_id = "f037e56e89293a057740de681ac9abbe"
+  list_id    = cloudflare_list.salesforce_ips.id
+  ip         = each.value
+}`,
+			},
+		},
+		{
+			Name: "dynamic block already with toset should not double wrap",
+			Config: `resource "cloudflare_list" "my_ips" {
+  account_id = "abc123"
+  name       = "my_ips"
+  kind       = "ip"
+
+  dynamic "item" {
+    for_each = toset(var.ip_addresses)
+    content {
+      value {
+        ip = item.value
+      }
+    }
+  }
+}`,
+			Expected: []string{
+				`resource "cloudflare_list" "my_ips" {
+  account_id = "abc123"
+  name       = "my_ips"
+  kind       = "ip"
+}`,
+				`resource "cloudflare_list_item" "my_ips_items" {
+  for_each   = toset(var.ip_addresses)
+  account_id = "abc123"
+  list_id    = cloudflare_list.my_ips.id
+  ip         = each.value
+}`,
+			},
+		},
+		{
+			Name: "dynamic block with item.key should use each.key",
+			Config: `resource "cloudflare_list" "asn_list" {
+  account_id = "abc123"
+  name       = "asn_list"
+  kind       = "asn"
+
+  dynamic "item" {
+    for_each = var.asn_map
+    content {
+      comment = item.key
+      value {
+        asn = item.value
+      }
+    }
+  }
+}`,
+			Expected: []string{
+				`resource "cloudflare_list" "asn_list" {
+  kind       = "asn"
+  account_id = "abc123"
+  name       = "asn_list"
+}`,
+				`resource "cloudflare_list_item" "asn_list_items" {
+  for_each   = toset(var.asn_map)
```

Comment:
humm, I think this is wrong. if you do `toset` the keys and values become identical. 

ie. iterating over `toset(["one", "two", "three"]) would yield

`each.key` -> `"one", "two", "three"`
and
`each.value` -> `"one", "two", "three"`

we want something like a map here where iterating over `for_each = {a="one", b="two"}`

would yield

`each.key` -> `"a", "b"`
and
`each.value` -> `"one", "two"`